### PR TITLE
chore: change accounts on twitter

### PR DESCRIPTION
**Pull request by guanbinrui**
closes #

### CODE REVIEW

File: packages/maskbook/src/extension/injected-script/etherumChange.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+              isMetaMask: boolean
+              on: (eventName: string, callback: (...args: unknown[]) => void) => void
+          })
+        | null
+}
+
+function registerListenersOnMetaMaskProvider() {
+    const ethereum = (globalThis as unknown as EthereumInjectedContext).ethereum
+    if (!ethereum?.isMetaMask) return
+    console.log('register listeners')
```
Comment (guanbinrui) [inline packages/maskbook/src/extension/injected-script/etherumChange.ts:16]:
```suggestion
```

File: packages/maskbook/src/extension/injected-script/etherumChange.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+function registerListenersOnMetaMaskProvider() {
+    const ethereum = (globalThis as unknown as EthereumInjectedContext).ethereum
+    if (!ethereum?.isMetaMask) return
+    ethereum.on('accountsChanged', onMetaMaskAccountsChange as (...args: unknown[]) => void)
+    ethereum.on('chainChanged', onMetaMaskChainIdChange as (...args: unknown[]) => void)
+}
+
+function onMetaMaskAccountsChange(accounts: string) {
+    console.log(`Accounts: ${accounts}`)
+    window.open('chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/index.html#/', '_blank', 'noopener noreferrer')
```
Comment (Jack-Works) [inline packages/maskbook/src/extension/injected-script/etherumChange.ts:22]: 🤔 can you explain this?

Comment (guanbinrui) [reply packages/maskbook/src/extension/injected-script/etherumChange.ts:22]: The users thought Mask Network connected theirs accounts with Twitter. So they switch accounts on the Twitter page. This handler would open the dashboard page if it detected the user switched their accounts in the wrong place. A dialog explains the right way of accounts management with MetaMask will be better.

Comment (Jack-Works) [reply packages/maskbook/src/extension/injected-script/etherumChange.ts:22]: But I don't see a URL like `index.html#/explain-metamask`. It seems like it jumps to the dashboard directly.

Comment (guanbinrui) [reply packages/maskbook/src/extension/injected-script/etherumChange.ts:22]: Actually, this URL is not allowed to open in this way. So I think an injected dialog will be better.

**Pull request closed without merging**